src/Microsoft.VisualStudio.LanguageServices.Razor/DefaultTagHelperResolver.cs

```diff
@@ -50,11 +50,12 @@ internal class DefaultTagHelperResolver : ITagHelperResolver
                                 // Per https://github.com/dotnet/roslyn/issues/12770 - there's currently no support for documentation in the OOP host
                                 // until that's available we add the documentation on the VS side by looking up each symbol again.
                                 var compilation = await project.GetCompilationAsync().ConfigureAwait(false);
-                                AddXmlDocumentation(compilation, result.Descriptors);
-                                return result;
+                                var documentedTagHelpers = GetDocumentedTagHelpers(compilation, result.Descriptors);
+                                var documentedResult = new TagHelperResolutionResult(documentedTagHelpers, result.Diagnostics);
```

Comment:
I'd suggest just writing it in a sane, albeit inefficient way.

1. Get the descriptors
2. Create the objects
3. Loop over each one that's missing docs and add them
4. return

Alternatively you could loop over the builders in stage 3.